### PR TITLE
Add Cauchy CDF scalar function

### LINKED ISSUE #4759: Implement cauchy_cdf prestosql function.
### Description

Add missing function to handle cauchy_cdf prestosql function.

### PR DESCRIPTION
**Pull request by gopukrishnasIBM**
Resolves #4759

### CODE REVIEW

**Comment by gopukrishnasIBM** [comment]
Here is the output of Fuzzer test.
```
./velox_expression_fuzzer_test  --seed 123456 --duration_sec 60 --logtostderr=1 --minloglevel=0  --only cauchy_cdf  --v=1 2>&1 | tee fuzzertest.out

I20230515 21:43:06.884850 22323706 ExpressionVerifier.cpp:68] All results match.
I20230515 21:43:06.884882 22323706 ExpressionFuzzer.cpp:1275] ==============================> Done with iteration 540
I20230515 21:43:06.885064 22323706 ExpressionFuzzer.cpp:1068] ==============================> Top 1 by number of rows processed
I20230515 21:43:06.885073 22323706 ExpressionFuzzer.cpp:1070] Format: functionName numTimesSelected proportionOfTimesSelected numProcessedRows
I20230515 21:43:06.885078 22323706 ExpressionFuzzer.cpp:1074] cauchy_cdf 57029 100.00% 13998
I20230515 21:43:06.885087 22323706 ExpressionFuzzer.cpp:1080] ==============================> Bottom 1 by number of rows processed
I20230515 21:43:06.885092 22323706 ExpressionFuzzer.cpp:1082] Format: functionName numTimesSelected proportionOfTimesSelected numProcessedRows
I20230515 21:43:06.885095 22323706 ExpressionFuzzer.cpp:1087] cauchy_cdf 57029 100.00% 13998
I20230515 21:43:06.885102 22323706 ExpressionFuzzer.cpp:1100] ==============================> All stats sorted by number of times the function was chosen
I20230515 21:43:06.885107 22323706 ExpressionFuzzer.cpp:1102] Format: functionName numTimesSelected proportionOfTimesSelected numProcessedRows
I20230515 21:43:06.885110 22323706 ExpressionFuzzer.cpp:1106] cauchy_cdf 57029 100.00% 13998
[==========] Running 0 tests from 0 test suites.
[==========] 0 tests from 0 test suites ran. (0 ms total)
[  PASSED  ] 0 tests.
```

File: velox/functions/prestosql/tests/ProbabilityTest.cpp [hunk 1]
```diff
@@ -155,5 +155,28 @@ TEST_F(ProbabilityTest, binomialCDF) {
...
+    return evaluateOnce<double>("cauchy_cdf(c0, c1, c2)", median, scale, value);
+  };
+
+  EXPECT_EQ(0.5, cauchyCDF(0.0, 1.0, 0.0));
+  EXPECT_EQ(0.75, cauchyCDF(0.0, 1.0, 1.0));
+  EXPECT_EQ(0.25, cauchyCDF(5.0, 2.0, 3.0));
+  EXPECT_EQ(1.0, cauchyCDF(5.0, 2.0, kInf));
+  EXPECT_EQ(0.5, cauchyCDF(5.0, kInf, 3.0));
+  EXPECT_EQ(0.0, cauchyCDF(kInf, 2.0, 3.0));
+  EXPECT_EQ(1.0, cauchyCDF(5.0, 2.0, kDoubleMax));
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/ProbabilityTest.cpp:171]: Any reason you didn't write a test where kDoubleMax is the first parameter ?

**Review by aditi-pandit — COMMENTED** [review]
Thanks for adding this function. This is looking good. Just have one comment.

**Comment by aditi-pandit** [comment]
@gopukrishnasIBM : There is a failure in the cauchyCDF unit test in linux-build. Please can you take a look.

File: velox/functions/prestosql/tests/ProbabilityTest.cpp (hunk 1 above)
Comment (gopukrishnasIBM) [reply velox/functions/prestosql/tests/ProbabilityTest.cpp:171]: @aditi-pandit No reason. Added in latest commit.

File: velox/functions/prestosql/tests/ProbabilityTest.cpp [hunk 2]
```diff
@@ -155,5 +155,29 @@ TEST_F(ProbabilityTest, binomialCDF) {
...
+  EXPECT_EQ(0.5, cauchyCDF(5.0, kDoubleMax, 3.0));
+  EXPECT_EQ(0.0, cauchyCDF(kDoubleMax, 1.0, 1.0));
+  EXPECT_EQ(0.75, cauchyCDF(kDoubleMin, 1.0, 1.0));
+  EXPECT_EQ(0.5, cauchyCDF(5.0, kDoubleMin, 5.0));
+  EXPECT_EQ(0.25, cauchyCDF(1.0, 1.0, kDoubleMin));
+  EXPECT_EQ(0.64758361765043326, cauchyCDF(2.5, 1.0, 3.0));
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/ProbabilityTest.cpp:177]: Please also add tests with kNan.

**Review by aditi-pandit — COMMENTED** [review]
@gopukrishnasIBM : The cauchyCDF test is failing in the linux-build. Please fix it.

**Comment by gopukrishnasIBM** [comment]
> @gopukrishnasIBM : The cauchyCDF test is failing in the linux-build. Please fix it.

@aditi-pandit linux-build failure is fixed.

**Review by aditi-pandit — APPROVED** [review]

**Pull request closed without merging**